unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/TextShape/Handler/ECSTextShapeComponentHandler.cs

```diff
@@ -176,18 +176,16 @@ internal void ApplyModelChanges(IDCLEntity entity, PBTextShape model)
             underlayKeywordEnabled = true;
         }
         
-        if (model.ShadowColor != null)
+        if (!underlayKeywordEnabled)
         {
-            if (!underlayKeywordEnabled)
-            {
-                textComponent.fontSharedMaterial.EnableKeyword("UNDERLAY_ON");
-                underlayKeywordEnabled = true;
-            }
-            var shadowColor  = new UnityEngine.Color(model.ShadowColor.R, model.ShadowColor.G, model.ShadowColor.B, model.Opacity);
-            textComponent.fontSharedMaterial.SetColor(underlayColor, shadowColor);
-            textComponent.fontSharedMaterial.SetFloat(offsetX, model.ShadowOffsetX);
-            textComponent.fontSharedMaterial.SetFloat(offsetY, model.ShadowOffsetY);
+            textComponent.fontSharedMaterial.EnableKeyword("UNDERLAY_ON");
+            underlayKeywordEnabled = true;
         }
+        var modelShadowColor = model.GetShadowColor();
+        var shadowColor  = new UnityEngine.Color(modelShadowColor.R, modelShadowColor.G, modelShadowColor.B, model.GetOpacity());
+        textComponent.fontSharedMaterial.SetColor(underlayColor, shadowColor);
+        textComponent.fontSharedMaterial.SetFloat(offsetX, model.ShadowOffsetX);
+        textComponent.fontSharedMaterial.SetFloat(offsetY, model.ShadowOffsetY);
```

Comment:
This has changed the functionality. We should check if the model has the shadow color to apply it. Otherwise, we are losing performance since we will setting it each time